### DISCUSSION TITLE
Failed storage object removal with functions in SQL Editor

### DISCUSSION THREAD

**Discussion by LanceNeedway** [supabase/supabase#18627] [Questions] [answered]
I’m using Flutterflow for my frontend and Subabase for my database.

I am able to upload user photots to a public “user” bucket under folder “pic.”
The url field in the “photo” column on my user profile table looks like this:
https://xnglwhbmvpvztkuqwheb.supabase.co/storage/v1/object/public/user/pic/1698760333849000.png

I am attempting to adapt the media management functions in the supabase tutorial here (https://supabase.com/docs/guides/getting-started/tutorials/with-nextjs?database-method=sql#set-up-the-database-schema) to my purposes.

However, I’m unable to delete storage objects with the following func:

```
create or replace function delete_storage_object_test(object text, out status int, out content text)
returns record
language 'plpgsql'
security definer
as $$
declare
  service_role_key text := '<keyFromMyProject>'; --  full access needed
begin
  select
      into status, content
           result.status::int, result.content::text
      FROM extensions.http((
    'DELETE',
    object,
    ARRAY[extensions.http_header('authorization','Bearer '||service_role_key)],
    NULL,
    NULL)::extensions.http_request) as result;
end;
$$;
```

Tested with the following query in SB SQL Editor where I pull the url from an existing file in the user/pic bucket as the function argument.

```
select
  delete_storage_object_test (
    'https://xnglwhbmvpvztkuqwheb.supabase.co/storage/v1/object/public/user/pic/1698769359593000.png'
  );
```

This test returns a failed delete request

(400,"{""statusCode"":""404"",""error"":""Not found"",""message"":""The resource was not found""}")

Is there something I’m missing about the way the storage api http extension works?

As I understand it the service role shouldn’t need any policies enabled. Are there specific policies I need to add to make this function work?

I am able to make a get request using a similar function.

**Comment by GaryAustin1** [comment]
There is/was an error in some of the guides around those functions.   https://github.com/orgs/supabase/discussions/17503#discussioncomment-7058356

**Reply by LanceNeedway** [reply]
Thanks for the reply! I did review and digested this particular issue already. For the sake of my own debugging process the problem I detailed bypasses the issue you referenced from that guide. I'm essentially just trying to get delete to work through the http extension in sql editor. So far I can't get it to work and I'm at a loss as to why.

**Reply by GaryAustin1** [reply, marked as answer]
You use public in your path.  Is that your bucket name?  If not and your bucket name is user then that is likely your error.  For public buckets that is only used in the PublicUrl.  All other operations have to meet RLS (you are security definer so do) AND use the basic path to the file which would not include the word public.

**Reply by LanceNeedway** [reply]
BINGO. Thanks a million! I removed public and it works as intended. Thanks for the assistance!
